### PR TITLE
Create a new resource to support attaching/detachign vpn to AWS TGW

### PR DESCRIPTION
**Pull request by zhiqiangjin**
resource_aviatrix_aws_tgw_vpn_conn. Support create/read/refresh/import/destroy.

### CODE REVIEW

File: aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+			"remote_cidr": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Remote CIDRs joined as a string with ','.",
+			},
+			"vpn_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the vpn generated by creation of the connection.",
```
Comment (arvindsree) [inline aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go]: say "vpn connection"

File: aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+		PublicIP:        d.Get("public_ip").(string),
+		RouteDomainName: d.Get("route_domain_name").(string),
+	}
+
+	remoteAsn := d.Get("remote_as_number").(string)
+	remoteCIDR := d.Get("remote_cidr").(string)
+
+	if remoteAsn != "" && remoteCIDR != "" {
+		return fmt.Errorf("remote_asn and remote_cidr cannot be both set. Please choose: " +
+			"remote_asw(dynamic) or remote_cidr(static)")
```
Comment (arvindsree) [inline aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go]: typo.. both should be remote_asn. also say "cannot be set at the same time"

File: aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	}
+
+	log.Printf("[INFO] Creating Aviatrix AWS TGW VPN Connection: %#v", awsTgwVpnConn)
+
+	err := client.CreateAwsTgwVpnConn(awsTgwVpnConn)
+	if err != nil {
+		return fmt.Errorf("failed to create Aviatrix AWS TGW VPN Connection: %s", err)
+	}
+
+	vpnID, err := client.GetAwsTgwVpnConnVpnId(awsTgwVpnConn)
```
Comment (arvindsree) [inline aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go]: move the logic to read

File: aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "This parameter represents the name of an AWS TGW.",
+			},
+			"route_domain_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of a route domain, to which the vpn will be attached.",
```
Comment (arvindsree) [inline aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go]: add default value as "Default Domain"

File: website/docs/r/aviatrix_aws_tgw_vpn_conn.html.markdown [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+   remote_cidr       = "16.0.0.0/16,16.1.0.0/16"
+ }
+ ```
+ 
+ ## Argument Reference
+ 
+ The following arguments are supported:
+ 
+ * `tgw_name` - (Required) This parameter represents the name of an AWS TGW.
+ * `route_domain_name` - (Required) The name of a route domain, to which the vpn will be attached.
```
Comment (arvindsree) [inline website/docs/r/aviatrix_aws_tgw_vpn_conn.html.markdown]: Mention that only Default_domain is supported now

File: aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go (hunk 4 above)
Comment (zhiqiangjin) [reply aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go]: If adding default value, it can not be required.

**Pull request merged**